hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+            for(char i : token){
+                my_token.push_back(i);
+            }
+            client_config->set_credentials(
+              std::make_shared<hazelcast::client::security::token_credentials>(
+                my_token));
+
+        }
+    }
+}
+void
+client_dom_config_processor::handle_serialization(
+  const boost::property_tree::ptree& node) const
+{
+    hazelcast::client::serialization_config serialization_config =
+      parse_serialization(node);
+    client_config->set_serialization_config(serialization_config);
+}
+void
+client_dom_config_processor::handle_network(
+  const boost::property_tree::ptree& node) const
+{
+    hazelcast::client::config::client_network_config client_network_config ;
+    for (auto& pair : node) {
+        auto child = pair_to_node(pair.first, pair.second);
+        std::string node_name = pair.first;
+        if (matches("cluster-members", node_name)) {
+            handle_cluster_members(child, &client_network_config);
+        } else if (matches("smart-routing", node_name)) {
+            client_network_config.set_smart_routing(
+              get_bool_value(pair.second.data()));
+        } else if (matches("redo-operation", node_name)) {
+            client_config->set_redo_operation(
+              get_bool_value(pair.second.data()));
+        } else if (matches("connection-timeout", node_name)) {
+            client_network_config.set_connection_timeout(
+              std::chrono::milliseconds(
+                get_integer_value(node_name, pair.second.data())));
+        } else if (matches("socket-options", node_name)) {
+            handle_socket_options(child, &client_network_config);
+        }  else if (matches("ssl", node_name)) {
+            handle_ssl_config(child, &client_network_config);
+        }else if (matches("hazelcast-cloud", node_name)) {
+            handle_hazelcast_cloud(child, &client_network_config);
+        } else if (matches("aws", node_name)) {
+            handle_aws(child, &client_network_config);
+        }
+    }
+    client_config->set_network_config(client_network_config);
+}
+
+void
+client_dom_config_processor::handle_cluster_members(
+  const boost::property_tree::ptree& node,
+  hazelcast::client::config::client_network_config* client_network_config)
+{
+    for (auto& pair : node) {
+        if (matches("address", pair.first)) {
+            auto pos = pair.second.data().find(':');
+            if(pos == std::string::npos){
+                client_network_config->add_address(
+                  hazelcast::client::address(pair.second.data(), 5701));
+                client_network_config->add_address(
+                  hazelcast::client::address(pair.second.data(), 5702));
+                client_network_config->add_address(
+                  hazelcast::client::address(pair.second.data(), 5703));
+                client_network_config->add_address(
+                  hazelcast::client::address(pair.second.data(), 5704));
```

Comment:
If port is not specified it should add 5701, 5702 5703 ports but not 5704 according to the documentation.